functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/jobs/ExpandedJobGroupsSpec.groovy

```diff
@@ -0,0 +1,150 @@
+package org.rundeck.tests.functional.selenium.jobs
+
+
+import org.rundeck.util.annotations.SeleniumCoreTest
+import org.rundeck.util.container.SeleniumBase
+import org.rundeck.util.gui.pages.home.HomePage
+import org.rundeck.util.gui.pages.jobs.*
+import org.rundeck.util.gui.pages.login.LoginPage
+import org.rundeck.util.gui.pages.project.ProjectCreatePage
+import org.rundeck.util.gui.pages.project.ProjectEditPage
+import spock.lang.Stepwise
+
+@SeleniumCoreTest
+@Stepwise
+class ExpandedJobGroupsSpec extends SeleniumBase {
+
+    def setup() {
+        def loginPage = go LoginPage
+        loginPage.login(TEST_USER, TEST_PASS)
+    }
+
+    /**
+     * It logs in as admin, creates a new project with the class name, empty description and empty label
+     * Then creates 4 jobs and groups them into different level layers
+     * Then goes to project settings, sets Job Group Expanded Levels to -1, 0, 1, 2 and validates Job Groups are expanded/hidden accordingly
+     * Then deletes the project and logs out
+     */
+
+    def "Job Groups are expanded or hidden accordingly"(){
+        when:
+        def expandedGroupsProp = "project.jobs.gui.groupExpandLevel="
+
+        String projectName = "ExpandedJobGroupsSpec"
+        def homePage = go HomePage
+        homePage.validatePage()
+
+        // Project create
+        def projectCreatePage = go ProjectCreatePage
+        projectCreatePage.validatePage()
+        projectCreatePage.createProject(projectName)
+        projectCreatePage.waitForUrlToContain("project/${projectName}/nodes/sources")
+
+        // Creates job 1
+        JobCreatePage jobCreatePage = page JobCreatePage
```

Comment:
you could use the API to create jobs. And use the UI only to setup the project configuration and validate accordingly